packages/peregrine/lib/talons/MiniCart/useEditItem.js

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from 'react';
+import { useQuery } from '@magento/peregrine';
+
+export const useEditItem = props => {
+    const { item, query } = props;
+    const [queryResult, queryApi] = useQuery(query);
+    const { data, error, loading } = queryResult;
+    const { runQuery, setLoading } = queryApi;
+
+    const itemHasOptions = item && item.options && item.options.length > 0;
+
+    // Run the query once on mount and again whenever the
+    // item being edited changes.
+    useEffect(() => {
+        const fetchItemVariants = async () => {
+            setLoading(true);
+
+            await runQuery({
+                variables: {
+                    name: item.name,
+                    onServer: false
+                }
+            });
+
+            setLoading(false);
```

Comment:
As mentioned in another PR, `setLoading(false)` is redundant here, since `runQuery` will call `receiveResponse`, prompting the reducer to set `loading` to `false`.

Reply:
I've done this elsewhere in other talons but I'll remove it here. Hopefully we can start using apollo hooks soon anyways.

Reply:
I still think it's weird we have to call `setLoading(true)` :D

Reply:
> I still think it's weird we have to call `setLoading(true)` :D

Yeah, it is weird. We only have to do it because the Apollo client is bugged, such that `runQuery()` never returns a response with `loading: true`. When we switch to official hooks, we shouldn't have to call `setLoading()` at all.